index.js

```diff
@@ -42,15 +48,28 @@ function List(/* items... */) {
 /* Creates a new list from the arguments (each a list item)
  * passed in. */
 function appendAll(list, items) {
-  var length = items && items.length
-  var index = -1
   var item
 
-  while (++index < length) {
-    item = items[index]
+  if (!items) {
+    return list
+  }
+
+  if (items[$iterator]) {
+    var iter = items[$iterator]()
+
+    while ((item = iter.next())) {
+      list.append(item.value)
+
+      if (item.done) {
+        break
+      }
+    }
+  } else {
+    var length = items.length
+    var index = -1
 
-    if (item !== null && item !== undefined) {
```

Comment:
This seems to remove support for ignoring holes in the list (`[item, , item]`, or `[item, null, item]`). Any reason for that?

Reply:
You already had support in `append`, didn't feel it was necessary to duplicate into both these places also.